### PR TITLE
python3Packages.mtcnn: init at 0.1.1; python3Packages.retinaface: init at 0.0.17; python3Packages.deepface: init at 0.0.92

### PR DESCRIPTION
**Pull request by DerDennisOP**
## Things done

New PR for #331179
This PR will work as soon the tensorflow package is fixed. #314235 

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/retinaface/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  };
+
+  postPatch = ''
+    # prevent collisions
+    substituteInPlace setup.py \
+      --replace-fail "data_files=[(\"\", [\"README.md\", \"requirements.txt\", \"package_info.json\"])]," "" \
+      --replace-fail "install_requires=requirements," ""
+  '';
+
+  disabledTestPaths = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/retinaface/default.nix:36]: We should add a comment why the tests are disabled

File: pkgs/development/python-modules/deepface/default.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  };
+
+  postPatch = ''
+    # prevent collisions
+    substituteInPlace setup.py \
+      --replace-fail "data_files=[(\"\", [\"README.md\", \"requirements.txt\", \"package_info.json\"])]," "" \
+      --replace-fail "install_requires=requirements," ""
+
+    substituteInPlace deepface/detectors/OpenCv.py \
+      --replace-fail "opencv_home = cv2.__file__" "opencv_home = os.readlink(cv2.__file__)" \
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/deepface/default.nix:45]: Would be nice to upstream this

File: pkgs/development/python-modules/deepface/default.nix [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  setuptools,
+  tensorflow,
+  tqdm,
+}:
+
+buildPythonPackage rec {
+  pname = "deepface";
+  version = "0.0.92";
+  pyproject = true;
+  build-system = [ setuptools ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/deepface/default.nix:27]:
```suggestion
```

File: pkgs/development/python-modules/deepface/default.nix [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      --replace-fail "data_files=[(\"\", [\"README.md\", \"requirements.txt\", \"package_info.json\"])]," "" \
+      --replace-fail "install_requires=requirements," ""
+
+    substituteInPlace deepface/detectors/OpenCv.py \
+      --replace-fail "opencv_home = cv2.__file__" "opencv_home = os.readlink(cv2.__file__)" \
+      --replace-fail "folders = opencv_home.split(os.path.sep)[0:-1]" "folders = opencv_home.split(os.path.sep)[0:-4]" \
+      --replace-fail "return path + \"/data/\"" "return path + \"/share/opencv4/haarcascades/\""
+  '';
+
+  propagatedBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/deepface/default.nix:50]:
```suggestion
  build-system = [ setuptools ];

  dependencies = [
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please run ``, nixfmt 'pkgs/development/python-modules/mtcnn/default.nix'``

File: pkgs/development/python-modules/deepface/default.nix (hunk 3 above)
Comment (DerDennisOP) [reply pkgs/development/python-modules/deepface/default.nix:27]: is there a reason to remove setuptools?

**Pull request merged**